src/web/browser/initPerf.ts

```diff
@@ -1,16 +1,27 @@
 export const initPerf = (name: string) => {
+    type TimeTakenInMilliseconds = number;
+
     const perf = window.performance;
     const startKey = `${name}-start`;
     const endKey = `${name}-end`;
 
     const start = () => {
         perf.mark(startKey);
     };
-    const end = () => {
+
+    const end = (dontLog: boolean = false): TimeTakenInMilliseconds => {
         perf.mark(endKey);
         perf.measure(name, startKey, endKey);
-        // eslint-disable-next-line no-console
-        console.log(JSON.stringify(perf.getEntriesByName(name)));
+
+        if (!dontLog) {
```

Comment:
I think this might be clearer without the double negative, so make it `shouldLog` with a default of `true` instead?

Reply:
I reckon that would be better, though it would change the interface of the function, which is used in quite a few places across DCR, and I'd have to update those to reflect the new interface. If you think this is a good idea, I'd be happy to do it - though it might be one for a follow up PR as it will touch quite a few files across DCR and is a fairly distinct piece of work.

Incidentally I think initPerf is responsible for quite a lot of console logging on production at the moment, it might not be a bad thing for the logging to be 'opt in' as it would make it a bit clearer that it's going to do that rather than tucking away the logging as a (potentially unintended) side effect.

Reply:
Agreed. It might be nicer for the start and end calls to return things so that the caller can decide whether they want to log, but I guess that's more a question for the platform team.

It comes down to what and who this initPerf module is really for. If it's going to have all these side-effects mixed in then it isn't very re-usable and I think we should consider calling the `window.performance` methods ourselves directly instead of adding abstraction on top of it to cater to our use-case.
As far as I can tell the module's value is pretty minimal right now, but maybe I'm missing some details or a future strategy reason for it? If the typical use-case is to call `record` and send the data to Ophan then should that be part of this module, for example?

Reply:
Interesting, what's your concern with logging to the console?

Reply:
I can see what you mean Adam - I had implemented my performance measurement separately at first but Tom had suggested I could reduce duplication by modifying initPerf instead as I was basically replicating all of it short of returning the value.

On logging - I can't speak for Adam but my thoughts were:

- I've found more 'noise' in the console can make it harder to use for development/debugging. You end up having to sort through irrelevant information to find what you've deliberately logged during the development process.
- It feels a bit of a shotgun approach - logging to every user's console every time they load a page, when probably what was needed was to log to a specific developer's device at a particular moment in time in order for them to observe the performance measurement. 
- If it's being used for performance monitoring I think the quality of analysis would be better if we were gathering performance information from real user devices and storing it somewhere to subsequently analyse it, rather than just logging it to the console and presumably reading specific instances on our own devices.

In the future we could read from somewhere like localStorage for a developer 'performance logging opt-in' before logging to the console.

Reply:
The marks are used within speedcurve already, via WPT which automatically recognises them. We can graph module load times over time with those metric. This is still synthetic and, you're right, it would be ideal to automatically capture those marks and measures and send them to ophan to get better RUM data.

Logging to user's consoles is not a huge problem, just because it will be thrown away if dev tools aren't open.

We don't need the console logging, but it's sometimes useful in production - that said, I've never used it in anger in production so I am happy to remove the console logging of these entirely. Dev tools will give us the same thing anyway.

Reply:
Thanks Gareth, useful to know! On your original suggestion - I've switched over to 'shouldLog' from the double negative now.

Reply:
Hey this feels like the one possibly unresolved thread on this PR. It feels like the three options are:

1) Remove the opt in/out of logging flag and continue to always log and not worry about it - we're already logging some performance metrics to the console and it seems like we're not worried about logging one additional metric.
2) Remove the console logging completely
3) Do nothing and merge as is

I think maybe I'd lean towards option 1 (followed closely by 2!) and have the conversation about what's useful to log in production separately.

Reply:
Thanks for taking another look Tom! I've followed number 1 and removed it - if we do feel the logging is a problem in the future we can sort it globally for DCR.